src/D2L.CodeStyle.Analyzers/ApiUsage/JsonParamBinderAttribute/JsonParamBinderAnalyzerFixer.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+
+namespace D2L.CodeStyle.Analyzers.ApiUsage.JsonParamBinderAttribute {
+
+	[ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(JsonParamBinderAnalyzerFixer) )]
+	public class JsonParamBinderAnalyzerFixer : CodeFixProvider {
+
+		public override FixAllProvider GetFixAllProvider() {
+			return WellKnownFixAllProviders.BatchFixer;
+		}
+
+		public override async Task RegisterCodeFixesAsync( CodeFixContext context ) {
+			Diagnostic diagnostic = context.Diagnostics.First();
+			TextSpan sourceSpan = diagnostic.Location.SourceSpan;
+
+			SyntaxNode root = await context.Document.GetSyntaxRootAsync(context.CancellationToken);

```

Comment:
`.ConfigureAwait( continueOnCapturedContext: false )`?

Reply:
Pretty sure it's not required since this is a GUI not a web app.
Checking though

Reply:
I don't know how Roslyn/VS sets their stuff up for analyzers. Regular GUIs definitely benefits from it because there is less thread contention.

Reply:
Used in the official docs by Roslyn devs. I've updated it to use ConfigureAwait(false)